### DISCUSSION TITLE
ASN.1 compiler?

### DISCUSSION THREAD

**Discussion by rainey19** [srsran/srsRAN_Project#55] [General Help] [answered]
Hello,

We're looking at implementing some rel. 17 features with SRSRan, and we need to update the asn1 definitions for RRC and F1AP to do so. What asn1 compiler did you use to generate them? I haven't found any references to one in the codebase yet, but I can't imagine writing 10k+ line structures by hand.

**Comment by andrepuschmann** [comment, marked as answer]
Our asn1 compiler is internal Software. However, in the next public release we will update our asn1 lib to release 17.

**Comment by orklann** [comment]
Hello, will you release the ans1 compiler as an open source project? or is it a commercial one?

**Reply by pgawlowicz** [reply]
We do not plan to release our asn1 compiler.

**Reply by frankist** [reply]
It's developed in-house, and we are considering releasing it. However, no date is yet decided.
